### PR TITLE
Logging Improvements

### PR DESCRIPTION
**Pull request by dommueller**
This pull requests fixes some of the existing logging issues.
* Fix possible memory leak in nextHead
* Directly stop logging, when stop() is called
* Restructure logging into public and private API (not 100% sure if everything is done correctly)
* The log buffer can now be resized and the initial log buffer size is set to 1000 (this slows down the performance for me in comparison but I guess this strongly depends on the system)
* Make logging configurable by adding a startExcept method

This is only tested on Mac OS X so far.

### CODE REVIEW

File: source/examples/comparison/main.cpp [hunk 1]
```diff
@@ -89,14 +89,14 @@ void compare()
     glbinding_error(false);
 
     std::cout << std::endl << "test: again, now with logging ..." << std::endl;
-    glbinding::logging::log(true);
+    glbinding::logging::start("logs/comparison");
```
Comment (scheibel) [inline source/examples/comparison/main.cpp]: The parameter ```filePath``` is treated as a base path where the timestamp is added?

Comment (dommueller) [inline source/examples/comparison/main.cpp]: No, it's the complete path. Therefore, if you run the comparison example multiple times only the last log file will persist and the "logs" directory will not be cluttered with hundreds of log files.

**Comment by scheibel** [comment]
The PR doesn't build on linux using gcc. Did you get an email?

**Comment by dommueller** [comment]
No I didn't get an email.

**Comment by dommueller** [comment]
Fixed the error and it should build now.

**Comment by scheibel** [comment]
At least, the pull request builder is working...

**Comment by dommueller** [comment]
Now I got an email saying it is back to normal. ;)

**Comment by scheibel** [comment]
Oh, so right now I shouldn't change anything on the build...let's hope I didn't just did that...

**Pull request merged**